### PR TITLE
Support passwords with spaces in them, fail on error

### PR DESCRIPTION
**Pull request by iddo**

### CODE REVIEW

File: src/backup [hunk 1]
```diff
@@ -18,6 +19,7 @@ echo 'creating backup archive of /var/www/html'
 tar --create --gzip -vv --directory="/var/www/html/" --file="/backups/backup_`date '+%Y%m%d'`.tar.gz" "./"
 
 echo 'creating database dump'
-mysqldump -h $MYSQL_ENV_MYSQL_HOST --add-drop-table -u$MYSQL_ENV_MYSQL_USER -p $MYSQL_ENV_MYSQL_DATABASE --password=$MYSQL_ENV_MYSQL_PASSWORD | bzip2 -c > /backups/backup_`date '+%Y%m%d'`.sql.bz2
+mysqldump -h "${MYSQL_ENV_MYSQL_HOST}" --add-drop-table --user="${MYSQL_ENV_MYSQL_USER}" -p "${MYSQL_ENV_MYSQL_DATABASE}" --password="${MYSQL_ENV_MYSQL_PASSWORD}" | bzip2 -c >
```
Comment (angelo-v) [inline src/backup]: Here is an invalid newline

**Review by angelo-v — CHANGES_REQUESTED** [review]
Thanks for your contribution. Nice catch! Could you remove the invalid newline in line 22? I will merge it then.

File: src/backup (hunk 1 above)
Comment (angelo-v) [reply src/backup]: Ah ok, could fix it myself :)

**Comment by angelo-v** [comment]
@iddo Oh and sorry, you need to change the restore command as well!

**Comment by iddo** [comment]
Fixed the newline, modified the restore script aswell

**Comment by iddo** [comment]
One thing I'm not sure of and haven't touched is the use of the `-p` flag, which is redundant when using `--password`.

I think it can be removed, but I'm not sure if it's there on purpose

**Comment by angelo-v** [comment]
> I think it can be removed, but I'm not sure if it's there on purpose

I think you are right, I will test it and remove it then

**Pull request merged**